src/Jobs/WorkflowTimeoutJob.php

```diff
@@ -21,7 +22,9 @@ public function getSignature()
 
     public function setup()
     {
-        $ttis = TimeoutTransitionInstance::get()->filter('Finished', false);
+        $curIds = WorkflowInstance::get()->filter('CurrentActionID:GreaterThan', 0)->column('CurrentActionID');
```

Comment:
This is just ensuring only 'future' transitions are being looped over and not already-executed ones yeah?

Reply:
Yep, filtering out any timeout's that aren't the current action on a workflow.